ledger/src/blockstore_db.rs

```diff
@@ -357,6 +357,18 @@ impl Rocks {
         Ok(opt)
     }
 
+    fn get_padded_cf(&self, cf: &ColumnFamily, key: &[u8], size: usize) -> Result<Option<Vec<u8>>> {
+        let opt = self.0.get_cf(cf, key)?.map(|db_vec| {
+            let mut bytes = vec![0; size];
+            {
+                let (data, _padding) = bytes.split_at_mut(db_vec.len());
+                data.copy_from_slice(&db_vec[..]);
+            }
+            bytes
+        });
+        Ok(opt)
+    }
```

Comment:
The reason to break this out into a separate function was because of `get_cf()`'s implementation:
```
fn get_cf(&self, cf: &ColumnFamily, key: &[u8]) -> Result<Option<Vec<u8>>> {
   let opt = self.0.get_cf(cf, key)?.map(|db_vec| db_vec.to_vec());
   Ok(opt)
}
```
Note the `db_vec.to_vec()`; [this clones db_vec](https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec.) into a new vector. If we do the resize later, then another allocation & memcpy will be necessary; this keeps us to one allocation & memcpy.